java/vector/src/main/java/org/apache/arrow/vector/compare/RangeEqualsVisitor.java

```diff
@@ -494,4 +505,46 @@ protected boolean compareFixedSizeListVectors(Range range) {
     }
     return true;
   }
+
+  protected boolean compareLargeListVectors(Range range) {
+    LargeListVector leftVector = (LargeListVector) left;
+    LargeListVector rightVector = (LargeListVector) right;
+
+    RangeEqualsVisitor innerVisitor =
+        createInnerVisitor(leftVector.getDataVector(), rightVector.getDataVector(), /*type comparator*/ null);
+    Range innerRange = new Range();
+
+    for (int i = 0; i < range.getLength(); i++) {
+      int leftIndex = range.getLeftStart() + i;
+      int rightIndex = range.getRightStart() + i;
+
+      boolean isNull = leftVector.isNull(leftIndex);
+      if (isNull != rightVector.isNull(rightIndex)) {
+        return false;
+      }
+
+      int offsetWidth = LargeListVector.OFFSET_WIDTH;
+
+      if (!isNull) {
+        final long startIndexLeft = leftVector.getOffsetBuffer().getLong(leftIndex * offsetWidth);
```

Comment:
Here we need to cast getLong argument to long.

Reply:
done